internal/resources/utils/clog/clog.go

```diff
@@ -0,0 +1,83 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package clog
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"go.uber.org/zap"
+)
+
+type Logger struct {
+	*logp.Logger
+}
+
+func (l *Logger) Errorf(template string, args ...any) {
+	// Downgrade context.Canceled errors to warning level
+	if hasErrorType(context.Canceled, args...) {
+		l.Warnf(template, args...)
+		return
+	}
+	l.Logger.Errorf(template, args...)
+}
+
+func (l *Logger) Error(args ...any) {
+	// Downgrade context.Canceled errors to warning level
+	if hasErrorType(context.Canceled, args...) {
+		l.Warn(args...)
+		return
+	}
+	l.Logger.Error(args...)
+}
+
+func (l *Logger) Named(name string) *Logger {
```

Comment:
I don't understand why we need to override this method (nor `WithOptions` and `With`)

Reply:
for transparency - the issue is the return type of `Named` (or other methods) is some struct `A` where we need it to be struct `B` (our logger, not the original one). this is required due to changes made in function signatures now expecting our custom logger and not the original one.

see https://goplay.tools/snippet/riQ9d7w6vOI for example